src/data/polynomial/algebra_map.lean

```diff
@@ -52,10 +53,31 @@ When we have `[comm_ring R]`, the function `C` is the same as `algebra_map R (po
 (But note that `C` is defined when `R` is not necessarily commutative, in which case
 `algebra_map` is not available.)
 -/
-lemma C_eq_algebra_map {R : Type*} [comm_semiring R] (r : R) :
+lemma C_eq_algebra_map (r : R) :
   C r = algebra_map R (polynomial R) r :=
 rfl
 
+variables (R)
+
+@[simp] lemma alg_hom_map_eq_C [algebra R A'] :
+  ↑(alg_hom_map R A' (polynomial A')) = (C : A' →+* polynomial A') := rfl
```

Comment:
Once you replace `alg_hom_map` with `is_scalar_tower.to_alg_hom` you'll find that `is_scalar_tower.coe_to_alg_hom` already simplifies the LHS.